src/python/pull_gsheets_translations_mvp.py

```diff
@@ -0,0 +1,136 @@
+
+languages = [x for x in translation_sheets.keys()]
+languages.sort()
+
+
+"""
+json structure
+{
+	"parentKey":{
+		"childKey":{
+			'fieldKey':{'value':'translatedValue','array':['translatedValue']},
+		}
+	},
+}
+"""
+
+def convert_to_camelCase(value):
+	#converts value to camelCase
+	camelCase = value.split()[0].lower() + " ".join(value.split()[1:]).title().replace(" ","")
+	text_tokens = word_tokenize(text)
+	#removes punctuation
+	return camelCase.translate(str.maketrans('', '', string.punctuation))
+
+import nltk
+nltk.download('punkt')
+nltk.download('stopwords')
+from nltk.tokenize import word_tokenize
+nltk.download('averaged_perceptron_tagger')
+
+def depunctuate(text):
+	chars = string.punctuation
+	for c in chars:
+		text = text.replace(c, "")
+	return text
+
+def education_value_cleaner(language_df):
+	lst = [word_tokenize(x) for x in language_df.value.tolist()]
+	#http://www.nltk.org/book_1ed/ch05.html
+	pos_tagged = [nltk.pos_tag(x) for x in lst]
+	lst_new=[]
+	for pt in pos_tagged:
+		nouns = [depunctuate(x[0]) for x in pt if "NN" in x[1]]
+		verbs = [depunctuate(x[0]) for x in pt if 'VB' in x[1]]
+		if len(verbs)==0 and len(nouns)==2:
+			lst_new.append(nouns[0].lower() + nouns[1].title())
+		elif len(verbs)==1 and len(nouns)==1:
+			lst_new.append(nouns[0].lower() + verbs[0].title())
+		else:
+			lst_new.append(nouns[0].lower() + nouns[1].title() + verbs[0].title())
+	return pd.Series(lst_new,index=language_df.index)
+
+for language in languages:
+	locale_key = [x for x in LANGUAGE_LETTERS_DICT.keys() if x in language][0]
+	locale = LANGUAGE_LETTERS_DICT[locale_key]
+	language_wks = translation_sheets[language]
+	wkDict = {}
+	for wk in language_wks:
+		wk_name = '_'.join(wk.title.split(' ')).replace('/','_')
+		language_lists = wk.get_all_values()
+		language_df = pd.DataFrame(language_lists[1:],columns=language_lists[0])
+		language_df = language_df[language_df['parentKey']!='']
+		# return a df with only the parentKey,fieldKey, childKey value, and translatedValue columns
+		language_df = language_df[['parentKey','fieldKey','childKey','value','translatedValue']]
+		# change value column to camelCase
+		if 'Education' in wk.title:
+			eduLanguage = language_df.copy()
+			language_df.value = education_value_cleaner(language_df)
+		else:
+			language_df['value'] = language_df['value'].apply(convert_to_camelCase)
+		parentKeyDict = {}
+		for parentKey, pgrp in language_df.groupby(['parentKey']):
+			childKeyDict ={}
+			for childKey,cgrp in pgrp.groupby(['childKey']):
+				fieldKeyDict = {}
+				for fieldKey,fgrp in cgrp.groupby(['fieldKey']):
+					fgrp_sub = fgrp.filter(regex='[Vv]alue')
+					valueDict = {"array":[]}
+					for value, translatedValue in fgrp_sub.values:
+						valueDict[value]=translatedValue
+						valueDict["array"].append(translatedValue)
+					fieldKeyDict[fieldKey] = valueDict
+				childKeyDict[childKey] = fieldKeyDict
+			parentKeyDict[parentKey] =  childKeyDict
+		wkDict.update(parentKeyDict)

```

Comment:
Is there no python/pandas/numpy function that would allow us to avoid this structure? If not, it's probably fine for the time being but this many nested for loops can lead to nasty worst case runtime complexity so it just stood out to me

Reply:
[this may be a good reference](https://stackoverflow.com/questions/42564432/optimizing-a-quadruple-nested-for-loop)

Reply:
Thanks for the reference! 

I’m going to look into the style for imports in a script and then do another commit. 

Right now the nested loops are needed. I was using pandas ‘to_json’ but it isn’t specific enough to produce both the array and value keys. I think we could do more effective list comprehensions but right now there shouldn’t be a concern for complexity since there’s so few rows in the gsheets. But definitely in the future we need better checks for a potential runtime error. 

Also is there a requirements.txt or something to require the python library dependencies?

Reply:
Fair enough - I figured this may be unavoidable for now, and even so it won't make much of a difference since `O(n^2)` vs `O(n^4)` is likely a matter of milliseconds when it's not thousands or hundreds of thousands of rows lol

And if you're asking if we can put in a `requirements.txt` I think we can just put it right in the `src/python` directory right? I honestly have very limited experience with python and have never used it in a professional setting. Feel free to add a `requirements.txt` to this PR, or if we want to do that in a separate PR at least add the required libs to the `README` for contributors to get up and running - good catch

Reply:
No great point - for scalability we need to reduce the complexity. But I’m not the best at doing so I’ll let a software engineer do that :)

I’m used to ‘pip install requirements.txt’ for getting libraries installed locally from modules or repos so I’ll include a requirements.txt in the PR soon. But again I do not know the most efficient way to mix js and python repos so in the future this should be amenable. Maybe we can have bc a feature enhancement issue?

Reply:
Yeah I'd say for now just adding the `requirements.txt` in `src/python` should be fine with no overlap 👍

Reply:
added it to the top of the dorectory - I don't ever see it in src/ folders

Reply:
I think it's better in `src/python` for organization

I think [pipenv](https://pipenv.pypa.io/en/latest/) has replaced `requirements.txt`, but we can handle that in a future issue/PR